blog/2023-06-22-openfeature-adoption-story.md

```diff
@@ -0,0 +1,129 @@
+Open Feature recently added [Events](https://github.com/open-feature/spec/blob/main/specification/sections/05-events.md#5-events). Together with provider events implementation, it could have given us this functionallity.  
+To overcome the missing events enhancement, we implemented it directly via provider abilities. It did cause some mix between OpenFeature and vendor specific implementation.
+
+## Initialization
+A common action is provider initialization. As it was vendor specific, it was done by provider call implementation.  
+Open Feature recently added [Initialization](https://github.com/open-feature/spec/blob/main/specification/sections/02-providers.md#24-initialization) to the spec. This could have made it a more standard implementation.
+
+## Fetching flags
+We used an option for fetching all enabled flags from the feature management service. This was not included in OpenFeature at the time, see example at [Flag Retrieval discussion](https://github.com/open-feature/ofep/issues/13#issuecomment-1337889563).
+
+## SDK used functionalities
+OpenFeature SDK has several used functionalities. What we used for our case:
+* Evaluation via OpenFeature Client (getBooleanEvaluation with context and similarly for other types)
+* Initialization - see [Initialization](#initialization)
+* Events - see [Monitoring](#monitoring)
+* Fetch all flags - see [Fetching flags](#fetching-flags)
+
+###  Features Management SDK - example flow
+
+```mermaid
+sequenceDiagram
+  participant SDK as SDK (server-side)
+  participant features as Features Service
+  actor request as User Request
+  participant developer as Developer
+  participant metrics as Metrics Provider
+  note left of SDK: initialize
+  rect rgb(30,144,255)
+  note left of SDK: OpenFeature method
+  end
+  SDK->>features: fetch config https://...
+  rect rgb(127,255,0)
+  features->>SDK: Feature Flags configuration JSON
+  end
+  note left of SDK: save config in cache
+  request->>SDK: getBooleanEvaluation
+  rect rgb(30,144,255)
+  note left of SDK: OpenFeature method
+  end
+  SDK->>request: true/false from cache
+  SDK->>features: <stream> fetch config changes events https://...
+  features->>SDK: OK
+  note left of features: fetch config changes responses
+  developer->>features: Change feature flag value
+  features->>SDK: Changed event
+  SDK->>features: fetch config https://...
+  rect rgb(255,0,0)
+  features->>SDK: Error
+  end
+  rect rgb(30,144,255)
+  note left of SDK: OpenFeature Error event
+  end
+  SDK->metrics: Features Service status=Error
+  request->>SDK: getBooleanEvaluation
+  SDK->>request: true/false from cache
+```
+
+## Summary
+
+### High Level Adoption
+* Main functionality used is for evaluating a feature flag value, which fit our needs.
+* Adopting OpenFeature flag required adding OpenFeature SDK dependency and using it.  
+  As no other major extra steps were needed, it is done via the adoption without containers related changes.
```

Comment:
> it is done via the adoption without containers related changes.

Could you please clarify what you mean?

Reply:
I am removing that sentence, less relevant for now. 
I meant that using OpenFeature did not require any devops/cluster changes like adding a new service/container.